### PR TITLE
add overall status for kubectl get cli

### PR DESCRIPTION
**Pull request by jinchihe**
**What this PR does / why we need it**:
Add overall status for `kubectl get kfservice` cli, see details in #306 

After the PR, the `kubectl get kfservice` will be shown as following:

```
[root@jinchi1 kfserving]# kubectl get kfservice
NAME             READY     URL                                          DEFAULT TRAFFIC   CANARY TRAFFIC   AGE
flowers-sample   Unknown   http://flowers-sample.kubeflow.example.com                                      43s
[root@jinchi1 kfserving]# kubectl get kfservice
NAME             READY     URL                                          DEFAULT TRAFFIC   CANARY TRAFFIC   AGE
flowers-sample   True      http://flowers-sample.kubeflow.example.com   90                10               3m
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/307)

### CODE REVIEW

**Comment by jinchihe** [comment]
/assign @yuzisun

As your comments in #306, that works fine. Thanks.

File: pkg/apis/serving/v1alpha2/kfservice_types.go [hunk 1]
```diff
@@ -149,6 +149,7 @@ type StatusConfigurationSpec struct {
 
 // KFService is the Schema for the services API
 // +k8s:openapi-gen=true
+// +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
```
Comment (yuzisun) [inline pkg/apis/serving/v1alpha2/kfservice_types.go]: checked knative they also print the reason field, what do you think adding that as well?
```
+kubebuilder:printcolumn:name="Reason",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].reason"
```

Comment (jinchihe) [reply pkg/apis/serving/v1alpha2/kfservice_types.go]: @yuzisun Personally I think we can just show overall status. some reasons: 1. User executes `kubectl get kfservice` just want to get the status (if ready or not), if add the reasons, may be too long for user quick checking. 2. If I were user, I would `kubectl describe` or `log` to check the details, and if have reasons, I guess that may be long.  3. Checked other CRD such as `tfjobs`, just shows status without reasons. :-) 
Of course, if you persist to that, I'm happy to add that. Thanks.

Comment (rakelkar) [reply pkg/apis/serving/v1alpha2/kfservice_types.go]: Is reason available on kfservice describe? I have found that I have to look at underlying configuration or revision or pod to figure out reason... not sure if this is related to your current change though?

Comment (ellis-bigelow) [reply pkg/apis/serving/v1alpha2/kfservice_types.go]: I agree with @jinchihe, it's already pretty long. I think the primary information is conveyed well, more details can always be retrieved without short output.

**Comment by ellis-bigelow** [comment]
/lgtm
Nice work! +1 for adding it in front of traffic, etc

File: pkg/apis/serving/v1alpha2/kfservice_types.go (hunk 1 above)
Comment (yuzisun) [reply pkg/apis/serving/v1alpha2/kfservice_types.go]: > Is reason available on kfservice describe? I have found that I have to look at underlying configuration or revision or pod to figure out reason... not sure if this is related to your current change though?

The reason should be already available on KFService status conditions, I am not very sure it shows up on `kubectl describe`.

**Pull request merged**